### ISSUE TITLE
results on tsplib

### ISSUE THREAD

**Issue by Wastedzz** [yining043/TSP-improve#4]
hi there, thanks for your contribution on this work, but the results of table 4 can't be reproduced, especially on AM(N=1280), AM(N=5000) and ours(T=3000), are you an co-author or a member of the institution mentioned in the original paper?

**Comment by yining043** [comment]
Hi, I am not the co-author of this paper but I am also doing research in this filed. May I know more details on what happened when you tried to reproduce table 4?

**Comment by Wastedzz** [comment]
> 
> 
> Hi, I am not the co-author of this paper but I am also doing research in this filed. May I know more details on what happened when you tried to reproduce table 4?

OK thanks again. When I reproduce the results in table 4, the objective values of these instances are quite larger than those reported in this table by a huge margin. The model of AM I used are the pretrained model: tsp_100 epoch-99.pt from Kool's github repository, the model of this paper is the one that I reproduce by your code.

**Comment by yining043** [comment]
Thanks for the reply. Based on my experience, may I check did you use the following normalization?

`data = (coordinates - np.min(coordinates)) / (np.max(coordinates) - np.min(coordinates))`

After getting the results, we need to scale back the obj values by multiplying the (np.max(coordinates) - np.min(coordinates)).

**Comment by Wastedzz** [comment]
> 
> 
> Thanks for the reply. Based on my experience, may I check did you use the following normalization?
> 
> `data = (coordinates - np.min(coordinates)) / (np.max(coordinates) - np.min(coordinates))`
> 
> After getting the results, we need to scale back the obj values by multiplying the (np.max(coordinates) - np.min(coordinates)).

yes, we did the same normalization but I didn't scale back because I find the AM model is better than paper's model after normalization

**Comment by yining043** [comment]
Oh, then I am not really sure about what causes such differences here.

**Comment by yining043** [comment]
Btw, I haven't tried to run this model or AM on TSPLIB but I tried to test the generalization of this model on different sizes (i.e., solve tsp100 with tsp50 model) which was found not good in my opinion.

**Comment by Wastedzz** [comment]
> 
> 
> Btw, I haven't tried to run this model or AM on TSPLIB but I tried to test the generalization of this model on different sizes (i.e., solve tsp100 with tsp50 model) which was found not good in my opinion.

you are right, I hold your opinion

**Comment by Wastedzz** [comment]
One more question, does or-tools have additional tricks during using? I use the [example ](https://github.com/google/or-tools/blob/stable/ortools/constraint_solver/samples/tsp_distance_matrix.py) version to run thes tsplib but the results are worse than those reported in this paper.

**Comment by yining043** [comment]
> One more question, does or-tools have additional tricks during using? I use the [example ](https://github.com/google/or-tools/blob/stable/ortools/constraint_solver/samples/tsp_distance_matrix.py) version to run thes tsplib instances but the results are worse than those reported in this paper.

Hmmm, I think the default setting should work well. Maybe you should check the hyper-parameters set for OR-TOOLS (e.g., the number of iterations?).
